### DISCUSSION TITLE
Redirect: False does not work. "Refreshes" my login page

### DISCUSSION THREAD

**Discussion by its-kos** [nextauthjs/next-auth#9322] [Help]
In my login page, I call "signIn" when the user submits the login form. For now, I want it to do nothing but console.log information just to see whats happening behind the scenes. However, every time I press the login button, the pages does a refresh. Upon inspecting the network tab I noticed the following 2 requests.

![image](https://github.com/nextauthjs/next-auth/assets/45661771/6f264227-ebca-442b-bc61-d2cf06198f4c)

Even If I remove the callback methods form my options, this refresh still happens. My login page is a custom one served at "/login"
This is the code that handles the login:
```
const handleLogin = (e: any) => {
        e.preventDefault()
        console.log("Login Page client: ", data)
        signIn("Credentials", {
            ...data,
            redirect: false
        })
}
```

**Comment by socketopp** [comment]
Hey, I just posted a similar thing. Have you found a way to mitigate this?

**Reply by its-kos** [reply]
No. Every research I've done, I find how to redirect instead of NOT redirect. I feel like no one has mentioned this issue anywhere. I'm not even sure if it's an issue or how the library is supposed to work.

**Reply by socketopp** [reply]
I just read some of the docs on [signIn ](https://next-auth.js.org/getting-started/client#using-the-redirect-false-option).
It says:
> The redirect option is only available for credentials and email providers.

I am using github as provider so maybe it's not possible.

I reckon you haven't found a way around this?

**Reply by its-kos** [reply]
The thing is I **don't** want to redirect. Everywhere I've found issues and solutions for how to redirect. Also I'm doing this with the credentials provider so no I have no idea how to solve this.

**Reply by socketopp** [reply]
I don't want to redirect either. I thought this was a way to enforce page transitions

**Reply by its-kos** [reply]
I'm not familiar with the library/framework enough to know more on this. So far however I haven't found a way to stop the redirect either way.

**Reply by socketopp** [reply]
Alright. I have found a way around redirect when it comes to my `signOut` case, updated #9407
Regarding your case, I would simply copy the signIn method for now do the implementation that works for you.
[However, inspecting the code, it seems that it should work as expected regarding redirect?](https://github.com/nextauthjs/next-auth/blob/main/packages/frameworks-sveltekit/src/lib/client.ts#L41-L95)
